spacelift/resource_stack_activator.go

```
@@ -0,0 +1,128 @@
+package spacelift
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/shurcooL/graphql"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/structs"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/validations"
+)
+
+func resourceStackActivator() *schema.Resource {
+	return &schema.Resource{
+		Description: "" +
+			"`spacelift_stack_activator` is used to to enable/disable Spacelift Stack.",
+		CreateContext: resourceStackActivatorCreate,
+		ReadContext:   resourceStackActivatorRead,
+		UpdateContext: resourceStackActivatorUpdate,
+		DeleteContext: schema.NoopContext,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"stack_id": {
+				Type:             schema.TypeString,
+				Description:      "ID of the stack to enable/disable",
+				Required:         true,
+				ValidateDiagFunc: validations.DisallowEmptyString,
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Description: "Enable/disable stack",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func resourceStackActivatorCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return resourceStackActivatorUpdate(ctx, d, meta)
+}
+
+func resourceStackActivatorRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	stack, err := queryStack(ctx, d, meta)
+	if err != nil {
+		return diag.Errorf("failed to retrieve stack: %v", internal.FromSpaceliftError(err))
+	}
+	if stack == nil {
+		d.SetId("")
```

Maybe we should return an error here if the stack can't be found?